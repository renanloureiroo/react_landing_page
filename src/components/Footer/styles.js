import styled, { css } from 'styled-components';

import { TextComponent } from '../TextComponent/index';
export const Container = styled.div`
  ${({ theme }) => css`
    & a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.sizes.small};
    }
    text-align: center;
  `}
`;

export const TextFooter = styled(TextComponent).attrs('div')``;
