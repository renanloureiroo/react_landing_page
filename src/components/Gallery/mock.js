export default {
  title: 'Gallery',
  description: 'Uma descrição sobre o conteúdo da minha galeria',
  images: [
    {
      id: 1,
      srcImg: 'https://source.unsplash.com/random/360x360?r=1',
      text: 'foto aleatória',
    },
    {
      id: 2,
      srcImg: 'https://source.unsplash.com/random/360x360?r=2',
      text: 'foto aleatória',
    },
    {
      id: 3,
      srcImg: 'https://source.unsplash.com/random/360x360?r=3',
      text: 'foto aleatória',
    },
    {
      id: 4,
      srcImg: 'https://source.unsplash.com/random/360x360?r=4',
      text: 'foto aleatória',
    },
    {
      id: 5,
      srcImg: 'https://source.unsplash.com/random/360x360?r=5',
      text: 'foto aleatória',
    },
    {
      id: 6,
      srcImg: 'https://source.unsplash.com/random/360x360?r=6',
      text: 'foto aleatória',
    },
  ],
};
