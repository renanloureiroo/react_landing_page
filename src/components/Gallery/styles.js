import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css``}
`;
export const TextContainer = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacing.huge};
  `}
`;
export const GridContainer = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    justify-content: center;
    gap: ${theme.spacing.medium};

    & div {
      width: 100%;
      height: 100%;
      margin: 0 auto;
      overflow: hidden;
    }

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;
export const Grid = styled.img`
  transition: all 300ms ease-in-out;
  ${() => css`
    width: 100%;
    height: 100%;

    &:hover {
      transform: scale(1.2) rotate(7deg);
    }
  `}
`;
