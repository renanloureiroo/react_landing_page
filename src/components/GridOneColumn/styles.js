import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    max-width: 50%;
    margin: 0 auto;
    text-align: center;
  `}
`;
