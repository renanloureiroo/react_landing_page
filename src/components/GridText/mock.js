export default {
  title: 'My Grid One',
  description: 'Uma breve descrição',
  bg: true,
  grids: [
    {
      title: 'Teste 1',
      description:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.',
    },
    {
      title: 'Teste 2',
      description:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.',
    },
    {
      title: 'Teste 3',
      description:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.',
    },
  ],
};
