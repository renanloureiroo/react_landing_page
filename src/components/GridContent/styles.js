import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    margin: 0 auto;
    max-width: 58rem;
    grid-template-columns: 1fr;
    text-align: center;
    justify-content: center;
    ${Title} {
      margin-bottom: ${theme.spacing.large};
    }
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
  & p{
    margin-bottom: ${theme.spacing.large}};
  }
  `}
`;
