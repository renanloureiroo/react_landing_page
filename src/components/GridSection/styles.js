import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-rows: 1fr 2fr;
    gap: ${theme.spacing.large};

    ${Title} {
      margin-bottom: ${theme.spacing.small};
    }

    @media ${theme.media.lteMedium} {
      display: block;
    }
  `}
`;
export const TextContainer = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacing.large};
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    gap: ${theme.spacing.medium};
    counter-reset: grid-counter;

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;
export const Grid = styled.div`
  ${({ theme }) => css`
    max-width: 36rem;
    ${Title} {
      position: relative;
      left: 5rem;
    }

    ${Title}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.spacing.xhuge};
      font-weight: bold;
      transform: rotate(10deg);
      left: -5rem;
      top: -2rem;
    }
  `}
`;
