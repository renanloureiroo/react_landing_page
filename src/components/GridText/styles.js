import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-rows: 1fr 2fr;
    gap: ${theme.spacing.large};

    ${Title} {
      margin-bottom: ${theme.spacing.small};
    }

    @media ${theme.media.lteMedium} {
      display: block;
    }
  `}
`;
export const TextContainer = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacing.large};
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(28rem, 1fr));
    gap: ${theme.spacing.large};
    counter-reset: grid-counter;

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;
export const Grid = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacing.medium};
    ${Title} {
      position: relative;
      left: 5rem;
    }

    ${Title}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.spacing.xhuge};
      font-weight: bold;
      transform: rotate(10deg);
      left: -5rem;
      top: -2rem;
    }
  `}
`;
