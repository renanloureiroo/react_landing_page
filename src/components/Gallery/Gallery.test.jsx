import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';
import { Gallery } from '.';

import mock from './mock';

describe('<Gallery />', () => {
  it('should render', () => {
    const { container } = renderTheme(<Gallery {...mock} />);
    expect(screen.getByRole('heading')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
