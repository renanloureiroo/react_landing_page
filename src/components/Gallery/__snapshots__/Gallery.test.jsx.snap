// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Gallery /> should render 1`] = `
.c2 {
  margin-bottom: 5.6rem;
}

.c5 {
  display: grid;
  grid-template-columns: repeat(3,1fr);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 2.4rem;
}

.c5 div {
  width: 100%;
  height: 100%;
  margin: 0 auto;
  overflow: hidden;
}

.c6 {
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  width: 100%;
  height: 100%;
}

.c6:hover {
  -webkit-transform: scale(1.2) rotate(7deg);
  -ms-transform: scale(1.2) rotate(7deg);
  transform: scale(1.2) rotate(7deg);
}

.c1 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background: #fff;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c4 {
  font-size: 1.6rem;
  line-height: 2.4rem;
}

@media (max-width:768px) {
  .c5 {
    grid-template-columns: 1fr;
  }
}

@media (max-width:768px) {
  .c3 {
    font-size: 4rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class=""
    >
      <div
        class="c2"
      >
        <h1
          class="c3"
        >
          Gallery
        </h1>
        <p
          class="c4"
        >
          Uma descrição sobre o conteúdo da minha galeria
        </p>
      </div>
      <div
        class="c5"
      >
        <div>
          <img
            alt="foto aleatória"
            class="c6"
            src="https://source.unsplash.com/random/360x360?r=1"
          />
        </div>
        <div>
          <img
            alt="foto aleatória"
            class="c6"
            src="https://source.unsplash.com/random/360x360?r=2"
          />
        </div>
        <div>
          <img
            alt="foto aleatória"
            class="c6"
            src="https://source.unsplash.com/random/360x360?r=3"
          />
        </div>
        <div>
          <img
            alt="foto aleatória"
            class="c6"
            src="https://source.unsplash.com/random/360x360?r=4"
          />
        </div>
        <div>
          <img
            alt="foto aleatória"
            class="c6"
            src="https://source.unsplash.com/random/360x360?r=5"
          />
        </div>
        <div>
          <img
            alt="foto aleatória"
            class="c6"
            src="https://source.unsplash.com/random/360x360?r=6"
          />
        </div>
        <div>
          <img
            alt="foto aleatória"
            class="c6"
            src="https://source.unsplash.com/random/360x360?r=7"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
